File: src/format.ts
import { extname } from "path";

export type Format = "markdown" | "json" | "yaml" | "po" | "text";

export function detectFormat(path: string): Format {
  const ext = extname(path).toLowerCase().replace(/^\./, "");
  switch (ext) {
    case "md":
    case "markdown":
      return "markdown";
    case "json":
      return "json";
    case "yaml":
    case "yml":
      return "yaml";
    case "po":
      return "po";
    default:
      return "text";
  }
}

File: internal/plan/format.go
package plan

import "strings"

type Format string

const (
	FormatMarkdown Format = "markdown"
	FormatJSON     Format = "json"
	FormatYAML     Format = "yaml"
	FormatPO       Format = "po"
	FormatText     Format = "text"
)

func DetectFormat(path string) Format {
	switch strings.ToLower(strings.TrimPrefix(ext(path), ".")) {
	case "md", "markdown":
		return FormatMarkdown
	case "json":
		return FormatJSON
	case "yaml", "yml":
		return FormatYAML
	case "po":
		return FormatPO
	default:
		return FormatText
	}
}

func ext(path string) string {
	idx := strings.LastIndex(path, ".")
	if idx == -1 {
		return ""
	}
	return path[idx:]
}
